pkg/node/manager/manager.go

```diff
@@ -385,9 +403,99 @@ func (m *manager) backgroundSyncInterval() time.Duration {
 	return m.ClusterSizeDependantInterval(baseBackgroundSyncInterval)
 }
 
+func (m *manager) carrierDownReconciler(ctx context.Context, health cell.Health) error {
```

Comment:
It would be good having a unit test covering this logic, to prevent accidentally breaking it in the future without noticing.